litellm/proxy/management_endpoints/key_management_endpoints.py

```diff
@@ -568,6 +568,16 @@ async def generate_key_fn(  # noqa: PLR0915
 
             data_json.pop("tags")
 
+        # --- TOKEN HANDLING LOGIC ---
+        if "token" in data_json and data_json["token"] is not None:
+            data_json["token"] = data_json["token"]
+        elif "key" in data_json and data_json["key"] is not None:
+            data_json["token"] = hash_token(data_json["key"])
+        else:
+            data_json["token"] = f"sk-{secrets.token_urlsafe(LENGTH_OF_LITELLM_GENERATED_KEY)}"
```

Comment:
it's not clear to me, why do we need this. I believe the code already generates the token if not provided

Reply:
You're right! All of this crap can be removed! 00e96f77f